src/stan/io/deserializer.hpp

```diff
@@ -0,0 +1,1978 @@
+   * bounds.
+   */
+  template <typename Ret, bool Jacobian, typename Offset, typename Mult,
+            typename LP, typename... Sizes,
+            require_std_vector_t<Ret>* = nullptr>
+  auto read_offset_multiplier(const std::vector<Offset>& offset,
+                              const Mult& multiplier, LP& lp,
+                              const size_t vecsize, Sizes... sizes) {
+    std::decay_t<Ret> ret;
+    ret.reserve(vecsize);
+    for (size_t i = 0; i < vecsize; ++i) {
+      ret.emplace_back(
+          this->read_offset_multiplier<value_type_t<Ret>, Jacobian>(
+              offset[i], multiplier, lp, sizes...));
+    }
+    return ret;
+  }
+
+  /**
+   * Specialization of offset multiplier for `std::vector` with
+   * non-`std::vector` offset and `std::vector` multiplier.
+   *
+   * <p>See <code>stan::math::offset_multiplier_constrain(T, double,
+   * double)</code>.
+   *
+   * @tparam Ret The type to return.
+   * @tparam Jacobian Whether to increment the log of the absolute Jacobian
+   * determinant of the transform.
+   * @tparam Offset Type of offset.
+   * @tparam Mult Type of multiplier.
+   * @tparam LP Type of log probability.
+   * @tparam Sizes A parameter pack of integral types.
+   * @param offset Offset.
+   * @param multiplier Multiplier.
+   * @param lp Reference to log probability variable to increment.
+   * @param vecsize The size of the return vector.
+   * @param sizes Pack of integrals to use to construct the return's type.
+   * bounds.
+   */
+  template <typename Ret, bool Jacobian, typename Offset, typename Mult,
+            typename LP, typename... Sizes,
+            require_std_vector_t<Ret>* = nullptr>
+  auto read_offset_multiplier(const Offset& offset,
+                              const std::vector<Mult>& multiplier, LP& lp,
+                              const size_t vecsize, Sizes... sizes) {
+    std::decay_t<Ret> ret;
+    ret.reserve(vecsize);
+    for (size_t i = 0; i < vecsize; ++i) {
+      ret.emplace_back(
+          this->read_offset_multiplier<value_type_t<Ret>, Jacobian>(
+              offset, multiplier[i], lp, sizes...));
+    }
+    return ret;
+  }
+
+  /**
+   * Return the next object, checking that it is a valid value for
+   * a probability, between 0 (inclusive) and 1 (inclusive).
+   *
+   * <p>See <code>stan::math::check_bounded(T)</code>.
+   *
+   * @tparam Ret The type to return.
+   * @tparam Sizes A parameter pack of integral types.
+   * @param sizes Pack of integrals to use to construct the return's type.
+   * @return Next probability value.
+   */
+  template <typename Ret, typename... Sizes>
+  auto read_prob(Sizes... sizes) {
+    auto ret = read<Ret>(sizes...);
+    using stan::math::check_bounded;
+    using stan::math::value_of;
+    check_bounded<decltype(value_of(ret)), double, double>(
+        "deserializer", "Constrained probability", value_of(ret), 0, 1);
+    return ret;
+  }
```

Comment:
I think this constraint also isn't used at the language level.

Reply:
Yeah I also couldn't find these in the stan manual? But I do know how to add these the stanc3 so if we wanted them idt it would be that bad to add

Reply:
Removing this also cause it's just `real<lower = 0.0, upper = 1.0> p;`